### ISSUE TITLE
App keeps opening in system browser

### ISSUE THREAD

**Issue by toniengelhardt** [GoogleChromeLabs/bubblewrap#549]
**Describe the bug**

I received the following review on Google Play:

![Screen Shot 2021-06-10 at 11 54 36](https://user-images.githubusercontent.com/2703233/121513904-48bc6f00-c9e3-11eb-862b-06b5869ede9e.png)

I think the issue might be related to the user having a default browser that is not PWA compatible or something like that.

**To Reproduce**

Can not reproduce. Other users with the same Android- and app version seem to have no problems.

**Expected behavior**

If the app says that it is available for the device in the PlayStore, it should always work as native experience.

**Smartphone**
 - Device: Meizu M5
 - Browser: unknown

**Comment by andreban** [comment]
AFAIK, Opera doesn't support Trusted Web Activity, so that may be the issue. This is the first time I hear about Flyme - I couldn't find it on Google Play. Does it support Trusted Web Activity/

**Comment by toniengelhardt** [comment]
I don't think so, it's also the first time I'm hearing about it. It seems to be the default browser of Meizu phones. What I'm a bit worried about is that users go to the PlayStore and download the app, but then it just doesn't provide a native experience if they don't have Chrome or Brave installed, which apparently leads to negative ratings.

I'm wondering if it would be possible to check if a TWA compatible browser is installed from the PlayStore and if not display a note saying "this app requires Chrome" or something, or not let the user install it at all, like when the app is not listed in your country?

**Comment by andreban** [comment]
> I'm wondering if it would be possible to check if a TWA compatible browser is installed from the PlayStore and if not display a note saying "this app requires Chrome" or something, or not let the user install it at all, like when the app is not listed in your country?

The TWA already [checks for a compatible browser](https://github.com/GoogleChrome/android-browser-helper/blob/main/androidbrowserhelper/src/main/java/com/google/androidbrowserhelper/trusted/TwaProviderPicker.java#L103-L172). If no compatible browser is found, there are [two fallback options](https://github.com/GoogleChromeLabs/bubblewrap/tree/main/packages/cli#:~:text=Defaults%20to%20true.-,fallbackType,-%27customtabs%27%20%7C%20%27webview%27):

- `customtabs`: Uses a [Custom Tab](https://developer.chrome.com/docs/android/custom-tabs/overview/) as the fallback and, ultimately, a standalone browser if one is not available.. ensures compatibility with the web platform and that the web content will work as expected, but shows the navigation bar (default).
- `webview`:  Uses a [WebView implementation](https://github.com/GoogleChrome/android-browser-helper/blob/main/androidbrowserhelper/src/main/java/com/google/androidbrowserhelper/trusted/WebViewFallbackActivity.java) as the fallback. It renders the content in fullscreen, but comes with the drawback that parts of the web app [may not work as expected](https://web.dev/web-on-android/#the-in-app-browser-challenge). I'd recommending testing your app using this fallback before deploying a version with the WV fallback enabled.
